File: answers/lintcode/Plus One.cpp
class Solution {
public:
    /**
     * @param digits a number represented as an array of digits
     * @return the result
     */
    vector<int> plusOne(vector<int>& digits) {
        // Write your code here
		int carry(1);
		for (size_t i = digits.size() - 1; i < digits.size(); -- i) {
			digits[i] += carry;
			carry = digits[i] / 10;
			digits[i] %= 10;
		}
		if (carry) {
			digits.insert(digits.begin(), carry);
		}
		return digits;
    }
};

File: answers/leetcode/Plus One/Plus One.py
class Solution(object):
    def plusOne(self, digits):
        """
        :type digits: List[int]
        :rtype: List[int]
        """

        carry = 1
        for i in reversed(range(len(digits))):
            carry += digits[i]
            digits[i] = carry % 10
            carry /= 10
        while carry:
            digits.insert(0, carry % 10)
            carry /= 10
        return digits
